runtime/common/declarationkind.go

```diff
@@ -190,3 +192,41 @@ func (k DeclarationKind) Keywords() string {
 func (k DeclarationKind) MarshalJSON() ([]byte, error) {
 	return json.Marshal(k.String())
 }
+
+var declarationKindMap = map[string]DeclarationKind{
+	"DeclarationKindUnknown":            DeclarationKindUnknown,
+	"DeclarationKindValue":              DeclarationKindValue,
+	"DeclarationKindFunction":           DeclarationKindFunction,
+	"DeclarationKindVariable":           DeclarationKindVariable,
+	"DeclarationKindConstant":           DeclarationKindConstant,
+	"DeclarationKindType":               DeclarationKindType,
+	"DeclarationKindParameter":          DeclarationKindParameter,
+	"DeclarationKindArgumentLabel":      DeclarationKindArgumentLabel,
+	"DeclarationKindStructure":          DeclarationKindStructure,
+	"DeclarationKindResource":           DeclarationKindResource,
+	"DeclarationKindContract":           DeclarationKindContract,
+	"DeclarationKindEvent":              DeclarationKindEvent,
+	"DeclarationKindField":              DeclarationKindField,
+	"DeclarationKindInitializer":        DeclarationKindInitializer,
+	"DeclarationKindDestructor":         DeclarationKindDestructor,
+	"DeclarationKindStructureInterface": DeclarationKindStructureInterface,
+	"DeclarationKindResourceInterface":  DeclarationKindResourceInterface,
+	"DeclarationKindContractInterface":  DeclarationKindContractInterface,
+	"DeclarationKindImport":             DeclarationKindImport,
+	"DeclarationKindSelf":               DeclarationKindSelf,
+	"DeclarationKindTransaction":        DeclarationKindTransaction,
+	"DeclarationKindPrepare":            DeclarationKindPrepare,
+	"DeclarationKindExecute":            DeclarationKindExecute,
+	"DeclarationKindTypeParameter":      DeclarationKindTypeParameter,
+	"DeclarationKindPragma":             DeclarationKindPragma,
+	"DeclarationKindEnum":               DeclarationKindEnum,
+	"DeclarationKindEnumCase":           DeclarationKindEnumCase,
+}
+
+func (a *DeclarationKind) UnmarshalJSON(b []byte) error {
+	if c, ok := declarationKindMap[strings.Trim(string(b), `"`)]; ok {
```

Comment:
Instead of trimming the serialized string's quotes manually, potentially leaving quoted content, unmarshal as a string:

```go
var s string
if err := json.Unmarshal(b, &s); err != nil {
	return err
}
```